starter-code/test/test.js

```diff
@@ -1,73 +1,72 @@
-// Require the file to be tested
-var SortedList = require('../index.js')
-
-// Load the Chai Assertion Library
-var assert = require('assert');
-
-describe('SortedList', function() {
-  describe('Constructor', function() {
-    it('should create an empty SortedList', function() {
-      var sl = new SortedList();
-      assert.equal(sl.length, 0);
+// Require the file which will be tested
+const SortedList = require('../index.js');
+// Load the Assertion Library
+const assert = require('assert');
+
+describe('SortedList', () => {
+  describe('Constructor', () => {
+    it('should have items and length properties', () => {
+      const sl = new SortedList();
       assert.equal(sl instanceof SortedList, true);
+      assert.equal(sl.length, 0);
+      assert(Array.isArray(sl.items))
     });
   });
 
-  describe('#add(x)', function() {
-    var sl;
-    beforeEach(function(){
+  describe('#add(x)', () => {
+    let sl;
+    beforeEach(() => {
       sl = new SortedList();
     });
-    it('should add a single value to SortedList', function() {
+    it('should add a single value to SortedList', () => {
       assert.equal(sl.length, 0);
       sl.add(1);
       assert.equal(sl.length, 1);
     });
-    it('should add a third value to SortedList', function() {
+    it('should add a third value to SortedList', () => {
       sl.add(30);
       sl.add(20);
       sl.add(10);
-      assert.equal(sl.length, 3);
+      assert.equal(sl.items.length, 3);
     });
   });
 
-  describe('#get(i)', function() {
-    var sl;
-    beforeEach(function(){
+  describe('#get(i)', () => {
+    let sl;
+    beforeEach(() => {
       sl = new SortedList();
-    });
-
-    it('should return an OutOfBounds exception if there is no element in that position', function() {
-      try {
-        sl.get(20);
-      } catch (e) {
-        assert.equal(e instanceof Error, true)
-        assert.equal(e.message, "OutOfBounds")
-        assert.throws(sl.get, Error, '/OutOfBounds/');
+      for(let i=0; i<3; i++) {
+        sl.add(i);
       }
+    })
+    
+    it('should return the element in that position', () => {
+      assert.equal(sl.get(2), 2);
     });
 
-    it('should return the element in that position', function() {
-      var foo = 10;
-      for(let i=0; i<200; i++) {
-        sl.add(foo*i);
-        assert.equal(sl.get(i), foo*i);
-      }
+    it('should return an OutOfBounds exception if there is no element in that position', () => {
+      assert.throws(() => {
+        if (sl.get(4) === 'undefined') {
+          throw new Error('OutOfBounds');
+        }
+      }, Error)
     });
+
+   
   });
 
-  describe('#add(x) and get(i)', function() {
-    var sl;
-    beforeEach(function(){
+  describe('#add(x) and get(i)', () =>  {
```

Comment:
these specs could be integrated in the unit that describes `add`

Reply:
This test requires using `add()` and `get()` but in the moment we define `add()`, we still don't have `get()` defined. That's why I kept it after. Maybe I am not thinking in the right direction. Please let me know if I am wrong. Thank you.

Reply:
The students implement `add()` without the sorting, and the tests pass. Then they don't necessarily see what was wrong with their initial implementation (does it come from the `add()` ? from the `get()` ?) and have to backtrack to implement the `add` method correctly.